packages/travix-ui-kit/components/slidingPanel/slidingPanelHeader/slidingPanelHeader.js

```diff
@@ -18,54 +18,107 @@ const renderDefaultLeftBlock = (backButtonLabel, onBackButtonClick) => (
   </button>
 );
 
-const SlidingPanelHeader = ({
+const leftBlock = (
   backButtonLabel,
-  children,
-  className,
-  dataAttrs,
-  leftBlock,
+  renderLeftBlock,
   onBackButtonClick,
-  rightBlock,
   useDefaultLeftBlock,
-}) => {
-  if (!children) {
-    return null;
+) => {
+  const defaultProps = {
+    className: "ui-sliding-panel-header__left-block-back",
+    onClick: onBackButtonClick,
+  };
+
+  if (renderLeftBlock) {
+    return renderLeftBlock(defaultProps);
   }
 
-  const headerClassName = classnames('ui-sliding-panel-header', className);
+  if (useDefaultLeftBlock) {
+    return renderDefaultLeftBlock(backButtonLabel, onBackButtonClick);
+  }
+
+  return null;
+};
 
-  const headerLeftBlock = useDefaultLeftBlock
-    ? renderDefaultLeftBlock(backButtonLabel, onBackButtonClick)
-    : leftBlock;
+const rightBlock = (renderRightBlock) => {
+  const defaultProps = {
+    className: "ui-sliding-panel-header__close-button",
+    'data-rel': "close",
+  };
 
-  const defaultCloseButton = (
-    <button
-      className="ui-sliding-panel-header__close-button"
-      data-rel="close"
-    >
+  if (renderRightBlock) {
+    return renderRightBlock(defaultProps);
+  }
+
+  return (
+    <button {...defaultProps}>
       &#215;
     </button>
   );
+};
 
-  return (
-    <div
-      className={headerClassName}
-      {...getDataAttributes(dataAttrs)}
-    >
-      <div className="ui-sliding-panel-header__left-block">
-        {headerLeftBlock}
-      </div>
-
-      <h3 className="ui-sliding-panel-header__title">
-        {children}
-      </h3>
+class SlidingPanelHeader extends Component {
+  componentWillMount() {
+    warnAboutDeprecatedProp(this.props.rightBlock, 'rightBlock', 'renderRightBlock property');
+    warnAboutDeprecatedProp(this.props.leftBlock, 'leftBlock', 'renderLeftBlock property');
+    warnAboutDeprecatedProp(
+      this.props.useDefaultLeftBlock,
+      'useDefaultLeftBlock',
+      'renderLeftBlock property and () => null'
+    );
```

Comment:
I'm not sure how we can describe a good replacement for `useDefaultLeftBlock`, as `renderLeftBlock={() => null}` would be equivalent to `useDefaultLeftBlock={false}`, but for `useDefaultLeftBlock={true}` it would be the default behavior, i.e. no prop at all.